Uno.Gallery/Uno.Gallery.Shared/Extensions/ControlExtensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using Windows.UI.Xaml;
+
+namespace Uno.Gallery
+{
+	/// <summary>
+	/// Helper class for Controls Extensions.
+	/// </summary>
+	public static class ControlExtensions
+	{
+		public static readonly DependencyProperty ResourcePathProperty =
+		DependencyProperty.RegisterAttached(
+			"ResourcePath",
+			typeof(string),
+			typeof(ControlExtensions),
+			new PropertyMetadata(null, OnResourcePathChanged));
+
+		public static void SetResourcePath(FrameworkElement element, string value)
+		{
+			element.SetValue(ResourcePathProperty, value);
+		}
+
+		public static string GetResourcePath(FrameworkElement element)
+		{
+			return (string)element.GetValue(ResourcePathProperty);
+		}
+
+		private static void OnResourcePathChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+		{
+			var control = d as FrameworkElement;
+			if (control != null)
+			{
+				var resourcePath = (string)e.NewValue;
+				AddResourcesFromPath(control, resourcePath);
+			}
+		}
+
+		private static void AddResourcesFromPath(FrameworkElement control, string resourcePath)
+		{
+			// Load the resource dictionary
+			var resources = new ResourceDictionary()
+			{
+				Source = new Uri(resourcePath, UriKind.RelativeOrAbsolute)
+			};
+
+			// Add or replace the resources in this control's resource dictionary
+			foreach (var key in resources.Keys)
+			{
+				if (control.Resources.ContainsKey(key))
+					control.Resources[key] = resources[key];
+				else
+					control.Resources.Add(key, resources[key]);
+			}
```

Comment:
you dont need to merge the resources, just replace the entire dictionary as suggested above

Reply:
and, it is not necessary to check `ContainsKey`, you can just do: dict[key] = value
for any generic dictionary, including res-dict